src/Service/HealthCheck/HttpUtilities.cs

```diff
@@ -104,11 +105,22 @@ public void ConfigureApiRoute(HttpContext httpContext)
                 {
                     // Send a GET request to the API
                     apiRoute = $"{apiRoute}{Utilities.CreateHttpRestQuery(entityName, first)}";
-                    HttpResponseMessage response = client.GetAsync(apiRoute).Result;
+                    HttpRequestMessage message = new(method: HttpMethod.Get, requestUri: apiRoute);
+                    if (string.IsNullOrEmpty(incomingRoleToken) && string.IsNullOrEmpty(incomingRoleHeader))
```

Comment:
Shouldnt this be checking for `!string.IsNullOrEmpty()` i.e. add the role header and token only if it is NOT empty?